MesaModel/model/data_types.py

```diff
@@ -2,25 +2,59 @@
 Global data types. These serve as containers for model parameters.
 """
 from dataclasses import dataclass
+from enum import Enum
 
 
 @dataclass(unsafe_hash=True)
 class GridParamType:
-    height: int
     width: int
+    height: int
+    n_groups: int
+    n_full_groups: int
+    n_group_cols: int
+    n_group_rows: int
+    group_width: int
+    group_height: int
 
 
 @dataclass(unsafe_hash=True)
```

Comment:
Given my comment above how would you feel about splitting this class so that the user parameters and the parameters for parameterisation were in separate dataclasses?

Reply:
I like this idea in theory, but I think there are some parameters that we might want in both places, e.g. teacher quality and control, which we need to parameterise, but also to allow a user to play around with to see the effect of improving.
Do you think it would work to have two types but still allow some to be user modifiable? Or possibly even three types (parameterisable, user-modifiable, both)?
Alternatively I guess we could accept a smaller number of parameters in run_pipeline and add the default values for the others, but that seems more liable to errors if we change the number of parameters later.

Reply:
Hmm interesting one. So as it stands we expose the user to some parameters in run_pipeline that they shouldn't touch but as a result the code is more easily maintained. Would we necessarily need to introduce a new dataclass? That is, could we create a wrapper around run_model_and_mlm() for parameterisation that reduces the number of params exposed to the user in the normal command line interface? That way we could retain one data structure which from the backend perspective makes sense - it is, I guess, irrelevant to the model itself whether a param is user defined or not. Maybe this is a separate issue for another PR?

Reply:
As discussed I've kept a single parameter type, but modified `run_pipeline.py` to only accept those parameters which we want to parameterise. I've added `STATIC_PARAMETER_COUNT` to `data_types.py` so that the last `STATIC_PARAMETER_COUNT` parameters are ones we should not be modifying as part of the parameterisation process. I hope that makes sense.